extension/src/cli/git/reader.ts

```diff
@@ -31,18 +31,28 @@ export class GitReader extends GitCli {
     return !!output
   }
 
-  public async getCommitMessage(cwd: string, sha: string) {
+  public async getLastThreeCommitMessages(cwd: string): Promise<{
+    [sha: string]: string
+  }> {
     const options = getOptions(
       cwd,
       Command.LOG,
       Flag.PRETTY_FORMAT_COMMIT_MESSAGE,
       Flag.NUMBER,
-      '1',
-      sha
+      '3',
```

Comment:
Not sure why but `dvc exp show -n 3` doesn't actually show the last three commits for our current demo project, but the 1st, 2nd, and 4th commits 🤔 

<img width="1248" alt="image" src="https://user-images.githubusercontent.com/43496356/210257760-3e4d51b1-c89b-4261-a1be-b21ebc293f56.png">

Reply:
looks like `exp show` is only showing merges and possibly the first parent:

```
git log --pretty="format:%H %B" HEAD~5..HEAD -z --merges --first-parent
```

This causes an issue where we cannot reliably grab the required commits by just going X commits back in time. We will need to pass the oldest commit into `git log` along with the required options. You should be able to get the commit by taking the last entry from `Object.keys` of `exp show` 😢.

Reply:
> You should be able to get the commit by taking the last entry from Object.keys of exp show 😢.

Good idea! I'll go ahead and increase the `-n` from 3 to 6 so there is less of a chance of it happening, merge, and fix it properly in a followup.